### PR TITLE
[`Docs`] Fix typos and update files at Optimization Page

### PR DESCRIPTION
**Pull request by standardAI**
# What does this PR do?

Fixed typos and trimmed (annoying) trailing whitespaces.

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [X] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu @yiyixuxu @DN6

### CODE REVIEW

File: docs/source/en/optimization/torch2.0.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
 specific language governing permissions and limitations under the License.
 -->
 
-# Torch 2.0
+# PyTorch 2.0
```
Comment (stevhliu) [inline docs/source/en/optimization/torch2.0.md:13]: Let's also update the `toctree` with this title so they're aligned.

**Review by stevhliu — COMMENTED** [review]
Haha thanks for cleaning up the whitespace! 🧹 

As with your other PR, let's hold off on making any changes to the Training docs since they'll be completely changed soon anyways when I finish updating them.

File: docs/source/en/optimization/coreml.md [hunk 2]
```diff
@@ -99,7 +98,7 @@ Once you have downloaded a snapshot of the model, you can test it using Apple's
 python -m python_coreml_stable_diffusion.pipeline --prompt "a photo of an astronaut riding a horse on mars" -i models/coreml-stable-diffusion-v1-4_original_packages -o </path/to/output/image> --compute-unit CPU_AND_GPU --seed 93
 ```
 
-`<output-mlpackages-directory>` should point to the checkpoint you downloaded in the step above, and `--compute-unit` indicates the hardware you want to allow for inference. It must be one of the following options: `ALL`, `CPU_AND_GPU`, `CPU_ONLY`, `CPU_AND_NE`. You may also provide an optional output path, and a seed for reproducibility.
```
[the remark this answers is not in the document]
Comment (standardAI) [reply docs/source/en/optimization/coreml.md:102]: Didn't "models/coreml-stable-diffusion-v1-4_original_packages" take the place of "<output-mlpackages-directory>"?

**Comment by standardAI** [comment]
I couldn't reverse new line additions at the end of the training files.

File: docs/source/en/optimization/coreml.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/optimization/coreml.md:102]: Yeah, what I mean is we can remove `<output-mlpackages-directory>` and instead say pass the path of the downloaded checkpoint to the script

**Review by stevhliu — APPROVED** [review]
Thanks for the fixes! You just need to rebase to get the CI to pass, and then we can merge 🙂

File: docs/source/en/optimization/coreml.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (standardAI) [reply docs/source/en/optimization/coreml.md:102]: WDYT now?

**Comment by standardAI** [comment]
Removed the training files in the PR.

File: docs/source/en/optimization/coreml.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/optimization/coreml.md:102]: LGTM! 👍

**Pull request merged**